provider/oauth2/views.py

```diff
@@ -22,12 +20,10 @@ class OAuth2AccessTokenMixin(AccessTokenMixin):
     def get_access_token(self, request, user, scope, client):
         try:
             # Attempt to fetch an existing access token.
-            at = AccessToken.objects.get(user=user, client=client,
-                                         scope=scope, expires__gt=now())
+            at = AccessToken.objects.get(user=user, client=client, scope=scope, expires__gt=now())
         except AccessToken.DoesNotExist:
             # None found... make a new one!
             at = self.create_access_token(request, user, scope, client)
-            self.create_refresh_token(request, user, scope, at, client)
```

Comment:
I see we are making 2 assumptions here:
1. All operators of open edX will never enable `constants.SINGLE_ACCESS_TOKEN`.
2. We will be removing this library in favor of django-oauth-toolkit so this is a temporary fix, anyway.

Not knowing what future lies ahead of us, in the meantime, why have broken functionality in this common library?  My concern is that (A) current (if any open edX instance has enabled constants.SINGLE_ACCESS_TOKEN) and (B) future uses of this library would not be doing the correct thing.

(By the way, the Mobile team was, in fact, considering to enable constants.SINGLE_ACCESS_TOKEN on edx.org since the access control table continues to grow in the multi-device and multi-session use cases.)

So either: (1) remove support for constants.SINGLE_ACCESS_TOKEN entirely - with notification to edx-code, (2) pass in a boolean to this method on whether or not a refresh_token is expected, or (3) refactor the abstraction as suggested above.  I believe, 2 or 3 should be reasonable to implement.  What do you say?